lib/pipelines.js

```diff
@@ -2,43 +2,41 @@
 const herokuAuthToken = require('./heroku-auth-token');
 const co = require('co');
 const shellpromise = require('shellpromise');
+const api = require('./heroku-api');
 const spawn = require('child_process').spawn;
 
-function api (url, token){
-	return fetch(
-		'https://api.heroku.com' + url,
-		{
-			headers: {
-				'Accept': 'Accept: application/vnd.heroku+json; version=3',
-				'Content-Type': 'application/json',
-				'Authorization': 'Bearer ' + token
-			}
-		}
-	).then(response => {
-		if(!response.ok){
-			let err = new Error(`BadResponse: ${response.status} ${response.statusText}`);
-			err.name = 'BAD_RESPONSE';
-			err.status = response.status;
-			throw err;
-		}
-
-		return response.json();
+async function info (pipelineName){
+	const authToken = await herokuAuthToken();
+	const pipeline = await api({
+		endpoint: `/pipelines/${pipelineName}`,
+		authToken
 	});
+
+	return pipeline;
 }
 
 function getApps (pipelineName){
 	return co(function* (){
-		let token = yield herokuAuthToken();
-		let pipelines = yield api('/pipelines', token);
+		let authToken = yield herokuAuthToken();
+		let pipelines = yield api({
```

Comment:
Doesn't need to be part of this PR - but we should consider refactoring this to use `async/await`.